From f9f5e9a0acafd97b13f89e8ac71c6ba9f18599ed Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Wed, 7 Sep 2016 16:28:45 -0700
Subject: [PATCH 1845/2748] CLOUDERA-BUILD. Excluding flaky test
 TestCapacityScheduler.testRecoverRequestAfterPreemption.

Fixing it requires pulling in YARN-4502, which brings in other potentially de-stabilizing scheduler changes. This is a lot of work, particularly given we deprecated CapacityScheduler.
Excluding the entire test would render the CapacityScheduler test less.

Change-Id: Ie95e1ec3bf08d000d2a4a83cfba5a119c7760276
---
 .../scheduler/capacity/TestCapacityScheduler.java  |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/TestCapacityScheduler.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/TestCapacityScheduler.java
index 6004993..af75876 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/TestCapacityScheduler.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/TestCapacityScheduler.java
@@ -1113,8 +1113,10 @@ public void testPreemptionInfo() throws Exception {
 
     rm1.stop();
   }
-  
-  @Test(timeout = 30000)
+
+  // Excluding this flaky test. Fixing it requires pulling in YARN-4502,
+  // which brings in other potentially de-stabilizing scheduler changes.
+  // @Test(timeout = 30000)
   public void testRecoverRequestAfterPreemption() throws Exception {
     Configuration conf = new Configuration();
     conf.setClass(YarnConfiguration.RM_SCHEDULER, CapacityScheduler.class,
-- 
1.7.9.5

